components/dashboard/src/dedicated-setup/use-check-dedicated-setup.ts

```diff
@@ -73,3 +69,26 @@ export const useOnboardingState = () => {
         },
     );
 };
+
+export const useNeedsSetup = () => {
```

Comment:
Extracted a separate hook from logic that was inside of `useCheckDedicatedSetup` to share it w/ Login view.